packages/core/src/application_init.ts

```diff
@@ -27,6 +27,80 @@ import {noop} from './util/noop';
  *
  * @see `ApplicationInitStatus`
  *
+ * @usageNotes
+ *
+ * ### Usage examples
+ *
+ * The following module provides a function returning a promise to the `APP_INITIALIZER` token.
+ *
+ * ```
+ *  function initializeApp(): Promise<any> {
+ *    return new Promise((resolve, reject) => {
+ *      // Do some asynchronous stuff
+ *      resolve();
+ *    });
+ *  }
+ *
+ *  @NgModule({
+ *   imports: [BrowserModule],
+ *   declarations: [AppComponent, HelloComponent],
+ *   bootstrap: [AppComponent],
+ *   providers: [{ provide: APP_INITIALIZER, useFactory: () => initializeApp }]
```

Comment:
I think we need to add `multi: true` here and in other examples, since `APP_INITIALIZER` is a token that represents a multi provider (also changed formatting a bit):
```suggestion
 *   providers: [{
 *     provide: APP_INITIALIZER,
 *     useFactory: () => initializeApp,
 *     multi: true
 *  }]
```